### PR TITLE
[Maintenance] Reorganize Config Files and Update Their Extensions

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
| --------------- | -----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no

### CODE REVIEW

File: docs/installation.md
Comment (Wojdylak) [inline docs/installation.md:1]: If the recipes work, you can remove this point "Add `FOS\RestBundle` configuration to your `config/packages/fos_rest.yaml` file"

File: tests/Application/config/routes/sylius_paypal.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 paypal_admin:
```
Comment (GSadee) [inline tests/Application/config/routes/sylius_paypal.yaml]:
```suggestion
sylius_paypal_admin:
```
and the same below

File: docs/installation.md
Comment (GSadee) [reply docs/installation.md:1]: We should separate the installation guide into the one with flex and without.

File: UPGRADE-2.0.md [hunk 2]
```diff
@@ -29,20 +29,20 @@
    ```diff
    sylius_paypal_shop:
    -   resource: "@SyliusPayPalPlugin/Resources/config/shop_routing.yaml"
-   +   resource: "@SyliusPayPalPlugin/config/shop_routing.yaml"
+   +   resource: "@SyliusPayPalPlugin/config/shop_routes.yaml"
```
Comment (diimpp) [inline UPGRADE-2.0.md]:
```suggestion
   +   resource: "@SyliusPayPalPlugin/config/routes/shop.yaml"
```

File: UPGRADE-2.0.md [hunk 3]
```diff
@@ -29,20 +29,20 @@
...
        prefix: /{_locale}
        requirements:
            _locale: ^[a-z]{2}(?:_[A-Z]{2})?$
 
    sylius_paypal_admin:
    -   resource: "@SyliusPayPalPlugin/Resources/config/admin_routing.yaml"
    -   prefix: /admin
-   +   resource: "@SyliusPayPalPlugin/config/admin_routing.yaml"
+   +   resource: "@SyliusPayPalPlugin/config/admin_routes.yaml"
```
Comment (diimpp) [inline UPGRADE-2.0.md]:
```suggestion
   +   resource: "@SyliusPayPalPlugin/config/routes/admin.yaml"
```

File: UPGRADE-2.0.md [hunk 4]
```diff
@@ -29,20 +29,20 @@
...
    +   prefix: '/%sylius_admin.path_name%'
 
    sylius_paypal_webhook:
    -   resource: "@SyliusPayPalPlugin/Resources/config/webhook_routing.yaml"
-   +   resource: "@SyliusPayPalPlugin/config/webhook_routing.yaml"
+   +   resource: "@SyliusPayPalPlugin/config/webhook_routes.yaml"
```
Comment (diimpp) [inline UPGRADE-2.0.md]:
```suggestion
   +   resource: "@SyliusPayPalPlugin/config/routes/webhook.yaml"
```

File: docs/legacy_installation.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        prefix: '/%sylius_admin.path_name%'
+    
+    sylius_invoicing_shop:
+        resource: '@SyliusPayPalPlugin/config/shop_routes.yaml'
+        prefix: /{_locale}
+        requirements:
+            _locale: ^[a-z]{2}(?:_[A-Z]{2})?$
+    ```
+
+1. Add `FOS\RestBundle` configuration to your `config/packages/sylius_paypal.yaml` file
```
Comment (diimpp) [inline docs/legacy_installation.md]: FOS rest configuration should be in `config/packages/fos_rest.yaml`, putting such "big" and common dependency into plugin is anti-pattern.

File: tests/Application/config/routes/sylius_paypal.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+sylius_paypal_admin:
```
Comment (diimpp) [inline tests/Application/config/routes/sylius_paypal.yaml]: Such common routes file should be at `config/routes.yaml` to allow end-users to import all routes at once.

File: UPGRADE-2.0.md (hunk 2 above)
Comment (GSadee) [reply UPGRADE-2.0.md]: We wanted to be consistent with other plugins, but doing it this way, as proposed, looks more like the Symfony way 🤔

File: tests/Application/config/routes/sylius_paypal.yaml (hunk 6 above)
Comment (GSadee) [reply tests/Application/config/routes/sylius_paypal.yaml]: It's only a test application, but are you suggesting to combine all plugin imports (admin, shop, webhook) into one `routes.yaml` and only import this file during installation into the end application?

Comment (diimpp) [reply tests/Application/config/routes/sylius_paypal.yaml]: >  only import this file during installation into the end application?

Yes, this way end-users who don't care for details will get full configuration without too much code on their side and still allow power end-users to import only shop/admin/webhook routes independently.

It's a kind of a good practice in public bundle configuration structure.

File: docs/legacy_installation.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    imports:
+        - { resource: '@SyliusPayPalPlugin/config/config.yaml' }
+    ```
+
+1. Import routes:
+
+    ````yaml
+   sylius_refund:
+       resource: "@SyliusPayPalPlugin/config/routes.yaml"
+    ````
```
Comment (GSadee) [inline docs/legacy_installation.md]:
```suggestion
    ```yaml
    sylius_refund:
        resource: "@SyliusPayPalPlugin/config/routes.yaml"
    ```
```

**Review by GSadee — APPROVED** [review]

**Pull request merged**